Python/chapter01/1.4 - PalinPerm/Nick4.py

```diff
@@ -15,15 +15,15 @@ def palperm(string):
         string (str): a string to check for all palindrome permutations
     '''
     table = {}
-    count = 0
     for letter in string.lower().replace(" ", ""):
         table[letter] = table.get(letter, 0) + 1
 
+    count = 0
```

Comment:
might be better to use [`dict.values()`](https://docs.python.org/3/library/stdtypes.html#dict.values):

```suggestion
    for value in table.values();
      if value % 2 == 1:
          return False
    return True
```

that can also be expressed in a one-liner with [`all()`](https://docs.python.org/3/library/functions.html#all):

```suggestion
    return all(value % 2 != 1 for value in table.values())
```

Reply:
@lhchavez Does also following follow the guard clause pattern ?

```suggestion
    odd_frequency_count = 0
    for frequency in letter_frequencies.values:
        if frequency % 2 == 1:
            odd_frequency_count += 1
            if odd_frequency_count > 1:
             return False
    return True
```

Reply:
you'd need to flip the condition:

```python
for frequency in letter_frequencies.values:
  if frequency % 2 == 0:
    continue
  odd_frequency_count += 1
  if odd_frequency_count > 1:
    return False
return True
```

an easy to remember heuristic is to try to minimize the indentation level. the real guidance is that the `if ` blocks (a.k.a. the guard clauses) should cover the non-main case and keep that logic out from the most important path. that way it's easy to recognize the different parts of the algorithm just based on visual inspection.